### PR TITLE
add lsp tests for some basic goto_def cases

### PR DESCRIPTION
**Pull request by Acepie**
This pr just adds unit tests to the existing language server implementation. Was originally thinking about looking into https://github.com/gleam-lang/gleam/issues/2655 but realized I probably need to read more/wanted to start with something smaller so just leaving the test cases here that I saw in the todos (hopefully helps the next person with implementing)

### CODE REVIEW

File: compiler-core/src/language_server/tests/definition.rs [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+            TextDocumentIdentifier::new(url),
+            position,
+        ),
+        work_done_progress_params: Default::default(),
+        partial_result_params: Default::default(),
+    };
+    let response = engine.goto_definition(params);
+
+    response.result.unwrap()
+}
```
Comment (lpil) [inline compiler-core/src/language_server/tests/definition.rs:41]: This looks familiar- is it copied from another test module? If so could we make that one public and use it across both?

**Review by lpil — COMMENTED** [review]
Amazing! Thank you! I've left one note inline

File: compiler-core/src/language_server/tests/definition.rs (hunk 1 above)
Comment (Acepie) [reply compiler-core/src/language_server/tests/definition.rs:41]: Good call, yeah this is effectively copied from hover (I thought I'd have to change it more at first). Lemme separate the engine create into a helper func

Comment (Acepie) [reply compiler-core/src/language_server/tests/definition.rs]: LMK if this is roughly what you had in mind

**Review by lpil — APPROVED** [review]
Thank you so much!

**Pull request merged**